Software/controller/controller.py

```diff
@@ -82,90 +83,97 @@ def on_message(client, userdata, msg):
-                            print("Occupied: " + str(place_number))
-                            self.model.list_of_places[place_number].occupyPlace()
-                            self.model.list_of_labels_to_display_place_number[place_number].config(
-                                bg="red")
-                            self.model.list_of_labels_to_display_ticket_number[place_number].config(
-                                text="Belegt")
-                            self.model.list_of_places[place_number].setstarttime(datetime.now(
-                                tz=None))
-                        else:
-                            print(
-                                "Not occupied - there is no ticket registered to this place")
-                    else:
-                        # Place was given up by owner
-                        print("Released: " +
-                              str(json_loaded["place_number"]))
-                        place_number = json_loaded["place_number"] - 1
-
-                        if self.model.list_of_places[place_number].state != PlaceState.OCCUPIED:
-                            print("Place is not in state OCCUPIED - can not be released")
-
-                        else:
-                            processing_time = datetime.now(
-                                tz=None) - self.model.list_of_places[place_number].start_time
-                            self.model.list_of_places[place_number] = Place(place_number)
-                            self.model.list_of_labels_to_display_place_number[place_number].config(bg="green")
-                            self.model.list_of_labels_to_display_ticket_number[place_number].config(
-                                text="Frei")
-                            # Save processing time in
-                            print("Processing time: " + str(processing_time))
-                            self.model.list_of_processing_times.append(processing_time)
-                            # Search for a new number in list_of_ticket_numbers to be registered to the released place
-                            if self.model.list_of_ticket_numbers:
-                                # ticket list is not empty - register number from ticket list to place
-                                # Take first element of list_of_ticket_numbers and register the number to the current place and display it
-                                ticket_number = self.model.list_of_ticket_numbers.pop(0)
-                                self.model.list_of_places[place_number].ticket_number = ticket_number
-                                self.model.list_of_places[place_number].state = PlaceState.REGISTERED
-                                self.model.list_of_labels_to_display_ticket_number[place_number].config(
-                                    text="%6d" % ticket_number)
-                                self.update_queue()
-
+                    handle_message_from_place(self, json_loaded)
```

Comment:
That should be `self.handle_message_from_place(json_loaded)`

(same for a few other calls)

Reply:
uhm, the function is defined within the function on_message - would that work here?

Reply:
defined functions within scope package, to comply to this